File: eNote/eNote.Model/DTOs/Predavanje.cs
namespace eNote.Model.DTOs
{
    public class Predavanje
    {
        public int Id { get; set; }
        public string Naziv { get; set; } = null!;
        public string Lokacija { get; set; } = null!;
        public int TrajanjeMinute { get; set; }        
        public string DatumVrijemePredavanja { get; set; } = null!;
        public string Kurs { get; set; } = null!;
        public string PredavanjeTip { get; set; } = null!;
        public string PredavanjeStatus { get; set; } = null!;
    }
}

File: eNote/UI/enote_desktop/lib/models/predavanje.dart
import 'package:json_annotation/json_annotation.dart';

part 'predavanje.g.dart';

@JsonSerializable()
class Predavanje {
  int? id;
  String? naziv;
  String? lokacija;
  String? datumVrijemePredavanja;
  int? trajanjeMinute;
  String? stanjePredavanja;
  int? kursId;
  String? kurs;
  String? tipPredavanjaId;
  String? tipPredavanja;

  Predavanje();

  factory Predavanje.fromJson(Map<String, dynamic> json) =>
      _$PredavanjeFromJson(json);

  Map<String, dynamic> toJson() => _$PredavanjeToJson(this);
}
